core/Types.h

```diff
@@ -277,8 +286,12 @@ class LambdaParam final : public Type {
     virtual TypePtr _instantiate(Context ctx, const InlinedVector<SymbolRef, 4> &params,
                                  const std::vector<TypePtr> &targs) override;
     virtual int kind() final;
+
+    bool isFixed() const {
+        return this->lower == this->upper;
```

Comment:
This relies on reference equality which is best effort

Reply:
OK, I'll go back to calling `isFixed` on the definition symbol then.